lib/aca_entities/atp/functions/build_application.rb

```diff
@@ -42,15 +43,44 @@ def application_hash
             us_state: 'ME', # default value
             assistance_year: 2021, # default_value,
             transfer_id: @memoized_data.resolve('external_id').item,
-            parent_living_out_of_home_terms: @memoized_data.resolve('parent_living_out_of_home_terms').item,
-            report_change_terms: @memoized_data.resolve('report_change_terms').item,
-            medicaid_terms: @memoized_data.resolve('medicaid_terms').item,
             is_renewal_authorized: @memoized_data.resolve('is_renewal_authorized').item,
             family_reference: { hbx_id: @memoized_data.resolve('family.hbx_id').item.to_s },
-            aptc_effective_date: Date.today # default value,
+            aptc_effective_date: Date.today, # default value
+            application_submission_terms: ssf_attestation_hash[:non_perjury_indicator],
+            medicaid_insurance_collection_terms: ssf_attestation_hash[:medicaid_obligations_indicator],
+            report_change_terms: ssf_attestation_hash[:report_change_terms],
+            years_to_renew: @memoized_data.resolve('insurance_application.coverage_renewal_year_quantity')&.item,
+            medicaid_terms: medicaid_terms_value,
+            parent_living_out_of_home_terms: parent_living_out_of_home_terms_value
           }
         end
 
+        def ssf_attestation_hash
+          @ssf_attestation_hash ||= begin
```

Comment:
Since we are calling the method multiple times but don't need to re-resolve the values we can use this handy multi-line memorization pattern. Its an optimization and not necessary so let me know if you still think we should remove it.

Reply:
Ah I see it now, nice work!